app/dockerdwrapper.c

```diff
@@ -153,6 +160,114 @@ get_parameter_value(const char *parameter_name)
   return parameter_value;
 }
 
+static bool
+create_dir_if_not_exists(char *base_path, const char *new_dir, mode_t mode)
+{
+  const char *combined_path = g_strdup_printf("%s/%s", base_path, new_dir);
+  syslog(LOG_INFO, "Create dir %s", combined_path);
+  int res = mkdir(combined_path, mode);
+  if (res != 0) {
+    if (EEXIST == errno) // TODO verify that the existing object is a directory
+                         // and set mode
+      res = 0;
+    else
+      syslog(LOG_ERR,
+             "Failed to create directory %s, Error code: %d",
+             combined_path,
+             errno);
+  }
+  return res == 0;
+}
+
+static bool
+move_file_or_symlink(const char *old_path, const char *new_path)
```

Comment:
Are you doing anything that 'mv' isn't doing? Could we just use 'mv'? Could we do this in the postinstall script instead?